File: app/src/main/java/org/netdex/androidusbscript/service/RootServiceConnection.java
package org.netdex.androidusbscript.service;

import android.content.ComponentName;
import android.content.ServiceConnection;
import android.os.IBinder;
import android.os.RemoteException;

import com.topjohnwu.superuser.nio.FileSystemManager;

public class RootServiceConnection implements ServiceConnection {
    private FileSystemManager remoteFS_ = null;

    @Override
    public void onServiceConnected(ComponentName name, IBinder service) {
        try {
            remoteFS_ = FileSystemManager.getRemote(service);
        } catch (RemoteException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onServiceDisconnected(ComponentName name) {
        remoteFS_ = null;
    }

    public FileSystemManager getRemoteFS() {
        return remoteFS_;
    }
}

File: app/src/main/java/org/netdex/androidusbscript/service/RootServiceConnection.kt
package org.netdex.androidusbscript.service

import android.content.ComponentName
import android.content.ServiceConnection
import android.os.IBinder
import android.os.RemoteException
import com.topjohnwu.superuser.nio.FileSystemManager
import timber.log.Timber

open class RootServiceConnection : ServiceConnection {
    var remoteFs: FileSystemManager? = null
        private set

    override fun onServiceConnected(name: ComponentName, service: IBinder) {
        try {
            remoteFs = FileSystemManager.getRemote(service)
        } catch (e: RemoteException) {
            Timber.e(e)
        }
    }

    override fun onServiceDisconnected(name: ComponentName) {
        remoteFs = null
    }
}
